app/assets/javascripts/thyme/thyme_player.js

```diff
@@ -0,0 +1,212 @@
+
+  // Update annotations after deleting an annotation
+  $(document).on('click', '#delete-button', function() {
+    const annotationId = Number(document.getElementById('annotation_id').textContent);
+    $.ajax(Routes.annotation_path(annotationId), {
+      type: 'DELETE',
+      dataType: 'json',
+      data: {
+        annotationId: annotationId
+      },
+      success: function() {
+        annotationManager.updateAnnotations();
+        // close and open again = show normal IA
+        iaButton.minus();
+        iaButton.plus();
+      }
+    });
+  });
+
+
+
+  /*
+    CHAPTERS & METADATA MANAGER
+   */
+  const iaBackButton = new IaBackButton('back-button', 'chapters');
+  iaBackButton.add();
+  const chapterManager  = new ChapterManager('chapters', iaBackButton);
+  const metadataManager = new MetadataManager('metadata');
+  thymeAttributes.chapterManager  = chapterManager;
+  thymeAttributes.metadataManager = metadataManager;
+  chapterManager.load();
+  metadataManager.load();
+
+
+
+  /*
+    INTERACTIVE AREA
+   */
+  iaButton.toHide = [$('#caption'), annotationArea];
+  iaButton.add();
+  (new IaCloseButton('ia-close', iaButton)).add();
+  //TODO
+  //const interactiveArea = new InteractiveArea();
+  //thymeAttributes.interactiveArea = interactiveArea;
+
+
+
+  /*
+    RESIZE
+   */
+  // Manage large and small display
+  function largeDisplayFunc() {
+    iaButton.plus();
+  }
+  const elements = [$('#caption'), $('#annotation-caption'), $('#video-controlBar')];
+  const displayManager = new DisplayManager(elements, largeDisplayFunc);
+
+  // resizes the thyme container to the window dimensions, taking into account
+  // whether the interactive area is displayed or hidden
+  function resizeContainer() {
+    const factor = $('#caption').is(':hidden') && $('#annotation-caption').is(':hidden') ? 1 : 1 / 0.82;
+    resize.resizeContainer(thymeContainer, factor);
+    if (thymeAttributes.annotations === null) {
+      annotationManager.updateAnnotations();
+    } else {
+      annotationManager.updateMarkers();
+    }
+  };
+  window.onresize = resizeContainer;
+  video.onloadedmetadata = resizeContainer;
+
+
+
+  /*
+    KEYBOARD SHORTCUTS
+   */
+  thymeKeyShortcuts.addGeneralShortcuts();
+  thymeKeyShortcuts.addPlayerShortcuts();
+
+
+
+  /*
+    MISC
+   */
+  const controlBarHider = new ControlBarHider('video-controlBar', 3000);
+  controlBarHider.install();
+  displayManager.updateControlBarType();
+  thymeUtility.playOnClick();
+  thymeUtility.setUpMaxTime('max-time');
+
+  // detect IE/edge and inform user that they are not suppported if necessary,
+  // only use browser player
+  if (document.documentMode || /Edge/.test(navigator.userAgent)) {
+    alert($('body').data('badbrowser'));
+    $('#caption').hide();
+    $('#annotation-caption').hide();
+    $('#video-controlBar').hide();
+    video.style.width = '100%';
+    video.controls = true;
+    resizeContainer();
+    window.onresize = resizeContainer;
+    return;
+  }
```

Comment:
Browser detection generally is considered bad practice, see the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent).

> Serving different Web pages or services to different browsers is usually a bad idea. The Web is meant to be accessible to everyone, regardless of which browser or device they're using. There are ways to develop your website to progressively enhance itself based on the availability of features rather than by targeting specific browsers.

Also note that the `navigator.userAgent` strings constantly changes. E.g. as of October 2023, the string for Edge is actually not "Edge", but "Edg" (see [here](https://stackoverflow.com/a/77234353/9655481) and I can confirm this on my machine).  Also see [this answer](https://stackoverflow.com/a/21712356/9655481) for how often the UserAgent string for Internet Explorer (IE) changed with time.

If one really wanted to detect the browser, one should use `navigator.userAgentData` and not the "legacy" `userAgent`, see [here](https://learn.microsoft.com/en-us/microsoft-edge/web-platform/user-agent-guidance).

---

All in all, I feel we should not bother with non-compliant browsers. If somebody is still using Internet Explorer these days to access MaMpf, we can't help it. For now, what about getting rid at least of the `/Edge/.test(navigator.userAgent)` part and file a future PR to get rid of IE support completely? If you agree, I could create a new issue so that we don't forget it.